controller/index.js

```diff
@@ -13,6 +12,24 @@ var Controller = function (agol, BaseController) {
   var controller = BaseController()
 
   /**
+  * Manages shared logic for any request that needs a host or key
+  *
+  * @param {object} req - the incoming request object
+  * @param {object} res - the outgoing response object
+  * @param {function} next - calls the next route handler
+  */
+  controller.main = function (req, res, next) {
```

Comment:
The two hard things in programming! Definitely agree that main is not so good. Maybe Controller.setHostKey? It's setting the cache key and looking up the AGOL host